derive/src/traits.rs

```diff
@@ -44,6 +44,11 @@ pub trait Derivable {
   fn explicit_bounds_attribute_name() -> Option<&'static str> {
     None
   }
+
+  /// If this trait has a custom meaning for "perfect derive", this function
+  /// should be overridden to return `Some`.
+  ///
+  /// The default is "the fields of a struct; unions and enums not supported".
```

Comment:
Good and important doc 👌 

  - 🤔 I wonder if we shouldn't be moving said default logic over here (by changing the output to `::syn::Result<Fields>`) to ensure actual code logic and this doc comment remain in sync